godot-codegen/src/godot_version.rs

```diff
@@ -72,23 +78,34 @@ fn test_godot_versions() {
         ("4.0.alpha.custom_build.faddbcfc0", 4, 0, 0, "alpha", s("faddbcfc0")),
         ("4.0.beta8.mono.custom_build.b28ddd918", 4, 0, 0, "beta", s("b28ddd918")),
         ("4.0.rc1.official.8843d9ad3", 4, 0, 0, "rc", s("8843d9ad3")),
+        ("4.0.stable.arch_linux", 4, 0, 0, "stable", None),
     ];
 
     let bad_versions = [
         "4.0.unstable.custom_build.e7e9e663b", // 'unstable'
         "4.0.3.custom_build.e7e9e663b",        // no stability
         "3.stable.official.206ba70f4",         // no minor
-        "4.0.alpha.custom_build",              // no rev after 'custom_build' (this is allowed for 'official' however)
```

Comment:
Yes, this removal is on purpose. I don't see why we need to reject it.

Reply:
See discussion in #118. Eventually we should probably accept them, but require manual mapping, warning, etc. Maybe the easiest approach for this would be parsing the version into something like:
```rs
enum GodotVersion {
    Release{major, minor, patch},
    Git{sha},
    Unknown
}
```

There's of course a separate discussion to be had whether we should still support `alpha/beta/rc` now, but at least `rc` can still occur before minor releases.